src/main/java/org/broadinstitute/hellbender/tools/exome/SegmentUtils.java

```diff
@@ -13,292 +15,452 @@
      */
-    public static double meanTargetCoverage(final Locatable loc, final TargetCollection<TargetCoverage> targets) {
-        Utils.nonNull(loc, "Can't get mean coverage of null segment.");
-        Utils.nonNull(targets, "Mean target coverage requires non-null targets collection.");
+    public static List<SimpleInterval> fixTargetSegmentStarts(final List<SimpleInterval> targetSegments,
+                                                              final TargetCollection<? extends Locatable> targets) {
+        Utils.nonNull(targetSegments, "The list of target segments cannot be null.");
+        Utils.nonNull(targets, "The collection of targets cannot be null.");
 
-        final List<TargetCoverage> myTargets = targets.targets(loc);
-
-        if (myTargets.size() == 0) {
-            throw new IllegalStateException("Empty segment -- no overlapping targets.");
+        final List<SimpleInterval> targetSegmentsFixed = new ArrayList<>();
+        for (final SimpleInterval segment : targetSegments) {
+            try {
+                final Locatable firstTarget = targets.targets(segment).get(0);
+                if (firstTarget.getEnd() == segment.getStart()) {
+                    targetSegmentsFixed.add(
+                            new SimpleInterval(segment.getContig(), firstTarget.getStart(), segment.getEnd()));
+                }
+            } catch (final IndexOutOfBoundsException ex) {
+                throw new IllegalArgumentException("List of targets and segments do not correspond.");
+            }
         }
-        return myTargets.stream().mapToDouble(TargetCoverage::getCoverage).average().getAsDouble();
+        return targetSegmentsFixed;
     }
-    
-    
+
     /**
-     *
-     * @param interval some genomic region
-     * @param targets collection of target genomic regions
-     * @param snps collection of SNP sites represented as length-1 intervals
-     *
-     * @return the smallest interval that contains the entirety (start to end) of every interval in {@code targets} and
-     * {@code snps} that overlaps {@code interval} at all.  If {@code interval} overlaps no targets or SNPs, return a length-1 interval
-     * at the start of {@code interval}.
-     *
-     * For example, if {@code interval} is 1:100-200 and the only target or snp it overlaps is 1:60-180, then the result
-     * is 1:60-180.
-     *
-     * Note that if {@code targets} contains overlapping targets we can get an inconsistent result where
-     * parsimoniousInterval returns an interval that is not parsimonious by its own definition.  For example, if {@code targets}
-     * contains 1:40-90 and 1:80-110 and {@code interval} is 1:100-200, then parsimoniousInterval yields 1:80-110, which
-     * falls in the middle of a target.  Since sites in {@code snps} are length-1, there is no such concern for them.
+     * Given an interval and collections of targets and SNPs, returns a trimmed interval produced by removing the empty
```

Comment:
that's right, because i think we want to avoid overlapping segments (which would occur if a SNP breakpoint falls within a target and we expanded intervals, for example).  i can add some documentation to clear this up.